src/transformers/models/gpt2/modeling_tf_gpt2.py

```diff
@@ -813,32 +813,29 @@ def get_output_embeddings(self):
     def set_output_embeddings(self, value):
         self.set_input_embeddings(value)
 
-    def prepare_inputs_for_generation(self, inputs, past=None, use_cache=None, use_xla=False, **kwargs):
-        # TODO: (Joao) after the TF generator is complete, update GPT2 TF generation to match PT's. NB -- some GPT2
-        # tests will need to be fixed after the change
-
+    def prepare_inputs_for_generation(self, inputs, past=None, use_cache=None, **kwargs):
+        token_type_ids = kwargs.get("token_type_ids", None)
```

Comment:
@gante 

We will need the same thing as
https://github.com/gante/transformers/blob/ed266959eb9f73d13661320d8534f9a5628b2fd0/src/transformers/generation_utils.py#L616-L619

for `_update_model_kwargs_for_generation` in `generation_tf_utils`, after allowing `token_type_ids` in `prepare_inputs_for_generation` here.

Otherwise, it won't either get the correct length (if `past` is `None`).